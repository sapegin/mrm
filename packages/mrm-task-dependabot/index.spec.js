jest.mock('fs');
jest.mock('git-username');
jest.mock('mrm-core/src/util/log', () => ({
	added: jest.fn(),
}));

const { getTaskOptions } = require('mrm');
const vol = require('memfs').vol;
const task = require('./index');

afterEach(() => vol.reset());

it('should add a workflow file', async () => {
	task(await getTaskOptions(task));

	expect(vol.toJSON()).toMatchSnapshot();

	// Ensure ok when file present
	task(await getTaskOptions(task));

	expect(vol.toJSON()).toMatchSnapshot();
});
