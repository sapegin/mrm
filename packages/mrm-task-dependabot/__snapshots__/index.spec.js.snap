// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add a workflow file 1`] = `
Object {
  "/.github/workflows/dependabot.yml": "name: Dependabot Automerge
on: pull_request
jobs:
  worker:
    runs-on: ubuntu-latest
    if: github.actor == 'dependabot[bot]'
    steps:
      - uses: actions/github-script@v3
        with:
          script: |-
            github.pulls.createReview({
              owner: context.payload.repository.owner.login,
              repo: context.payload.repository.name,
              pull_number: context.payload.pull_request.number,
              event: 'APPROVE'
            })
            github.pulls.merge({
              owner: context.payload.repository.owner.login,
              repo: context.payload.repository.name,
              pull_number: context.payload.pull_request.number,
              merge_method: 'squash'
            })
",
}
`;

exports[`should add a workflow file 2`] = `
Object {
  "/.github/workflows/dependabot.yml": "name: Dependabot Automerge
on: pull_request
jobs:
  worker:
    runs-on: ubuntu-latest
    if: github.actor == 'dependabot[bot]'
    steps:
      - uses: actions/github-script@v3
        with:
          script: |-
            github.pulls.createReview({
              owner: context.payload.repository.owner.login,
              repo: context.payload.repository.name,
              pull_number: context.payload.pull_request.number,
              event: 'APPROVE'
            })
            github.pulls.merge({
              owner: context.payload.repository.owner.login,
              repo: context.payload.repository.name,
              pull_number: context.payload.pull_request.number,
              merge_method: 'squash'
            })
",
}
`;
